File: 214/projects/08/java/TempTester.java
/* TempTester.java is a 'driver' to test class Temp.
 * Prof: Dr. Adams, for CS 214 at Calvin College.
 * Student: Chan Kim (ck45)
 * Date: 4/9/2018
 ************************************************************/

import java.io.*;
import java.util.Scanner;

public class TempTester  
{
  public static void main(String[] args)
  {
    Scanner keyboard = new Scanner(System.in);
    
    Temp baseTemp = new Temp(0.0, 'C');
    baseTemp.read(keyboard);
    Temp limitTemp = new Temp(0.0, 'C');
    limitTemp.read(keyboard);

    double stepValue = keyboard.nextDouble();
    
    char scale = baseTemp.getScale();

    while (baseTemp.equals(limitTemp) || baseTemp.less_than(limitTemp))
    {
      baseTemp.setFahrenheit();
      System.out.print(baseTemp.toString() + "\t");
      baseTemp.setCelsius();
      System.out.print(baseTemp.toString() + "\t");
      baseTemp.setKelvin();
      System.out.print(baseTemp.toString() + "\n");

      if (scale == 'F')
      {
        baseTemp.setFahrenheit();        
      }
      else if (scale == 'C')
      {
        baseTemp.setCelsius();
      }
      else if (scale == 'K')
      {
        baseTemp.setKelvin();
      }

      baseTemp.raise(stepValue);
    }
  }
}

File: 214/projects/08/ruby/tempTester.rb
# tempTester.rb tests class Temp and its operations
#
# Prof: Dr. Adams, for CS 214 at Calvin College.
# Student: Chan Kim (ck45)
# Date: 4/9/2018
####################################################

require './Temp.rb'

def testTemp
        baseTemp = Temps::Temp.new(0.0, 'C')
        baseTemp.read
        limitTemp = Temps::Temp.new(0.0, 'C')
        limitTemp.read
        
        stepValue = gets.to_f

        scale = baseTemp.scale

        while (baseTemp.equals(limitTemp) && baseTemp.less_than(limitTemp)) do
                
                baseTemp.setFahrenheit
                baseTemp.write
                print("\t")
                baseTemp.setCelsius
                baseTemp.write
                print("\t")
                baseTemp.setKelvin
                baseTemp.write
                print("\n")

                if (scale == 'F') then
                        baseTemp.setFahrenheit
                elsif (scale == 'C') then
                        baseTemp.setCelsius
                elsif (scale == 'K') then
                        baseTemp.setKelvin
                end

                baseTemp.raise(stepValue)
        end
end

testTemp

